[4.0] fix webp support

brianteeman opened the pull request:
#31048 added support for webp images however it was not complete. MediaHelper::isImage was not updated to support webp. As a result while webp could be added to media manager as an allowed image format everything worked except for the display of the thumbnail.

As this PR fixes that we might as well add webp to the default list of supported image formats.

### To test

1. either do a fresh install with the prebuilt package OR apply this pr, run npm ci and go to the media manager options. delete all the values in the Legal Extensions (File Types), Legal Image Extensions (File Types) and Legal MIME Types and then save the options. This will reset the value to the new defaults.
2. Try to upload any webp image
3. Try to crop and resize the image
4. Try to rename the image
5. Get a shareable link and make sure it works

### Before
![image](https://user-images.githubusercontent.com/1296369/114272607-be6d8400-9a0e-11eb-9217-ef2dbe8cfbe3.png)

### After
![image](https://user-images.githubusercontent.com/1296369/114272628-cdeccd00-9a0e-11eb-8c91-875207a7b7df.png)

Commit:
s/ordern/order (#32565)

Commit:
typo - Add a space (#32580)

Commit:
s/Append/Prepend (#32570)

Commit:
s/carefull/careful (#32575)

merged on review

rdeutz committed:
remove deprecated method (#32356)

* remove deprecated method

* remove getCfg calls

* Simplify code since every Joomla app has a get method

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Commit:
s/1.6.0/3.9.25 (#32581)

richard67 committed:
[3.x] Make database schema check ignore display widths for all integer types on MySQL or MariaDB databases (#32605)

* Fix and simplify display width for MySQL change item

* Fix type check for check query, too

* Refactoring type check

* Fix comments for regex's

* Fix PHPCS 1st trial

joomdonation committed:
Fix submit article permission (frontend) (#32470)

joomdonation committed:
Fix select Group Parent on group creation error (#32341)

saharin88 committed:
link to a field outside of any group (#32646)

merged on review, was already RTC as another PR

weeblr committed:
Menus do not comply with access levels set when logging-in through re… (#30991)

* Menus do not comply with access levels set when logging-in through remember me feature

joomdonation committed:
Fix Privacy Request Email Data Export Error (#32299)

* Fix Privacy Request Email Data Export Error

* Update administrator/components/com_privacy/views/request/view.html.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Commit:
s/404/500 (#32579)

Commit:
[3] Add passing unit test for FilePathRule (#32723)

* Add unit test for FilePathRule

HLeithner committed:
Fix Joomla version note (#32731)

richard67 committed:
Fix b/c issue with folders for files and images of com_media from the 3.9.25 release (#32718)

* Fix FilePathRule b/c issue

* Exclude list cannot be empty now

* Adapt new unit tests from PR 32723 to the changes of this one here

* Add test cases for excluded core folders

* Add exclude parameter to the form field for unit tests

Fedik committed:
Calendar change event (#32756)

rdeutz committed:
Update README.md

joomdonation committed:
Fix warnings on php 8 (#32767)

joomdonation committed:
Fix category created/modified date (#32857)

GeraintEdwards committed:
Pre-Upgrade Compatibility Check on Plugins That Could Kill Joomla Upgrade (#31200)

* Alternative presentation of compatibility checks for Joomla 4 from Joomla 3.10

* Alternative presentation of compatibility checks for Joomla 4 from Joomla 3.10

* remove accidental commit of .gitignore

* Handle situation where we are running 3.10 with an extension what doesn't say its compatible with 3.10 - was incorrectly being reported as not supporting Joomla updater

* Adding user guidance notes for update check

* code style

* Update administrator/components/com_joomlaupdate/controllers/update.php

Co-authored-by: Quy <quy@fluxbb.org>

* Update administrator/components/com_joomlaupdate/controllers/update.php

Co-authored-by: Quy <quy@fluxbb.org>

* Changed styling of compatibility feedback columns so that they are aligned
Make warning message display be conditional on being relevant to the displayed output.

* Revert accidentally committed file

* Use tabs for HTML spacing
Alphabetical language string key definition
Resolve undeclared updateInfo['current']

* Missing ->state when checking "if ($currentCompatibilityStatus->state == 1)"

* implement php cs changes

* Changes suggested by zero-24
* change updating icon to Joomla ajax-loader.gif
* Styling the results with margin left and right
Also corrected language type Jomla instead of Joomla

* Revert accidentally committed file

* More changes suggested by zero-24
* Remove double spacing after full stop in language strings

* Changes suggested by zero-24
'is not supported' - awaiting feedback on capitalisation

* Changes suggested by zero-24
* js file spacing

* Changes suggested by zero-24
* Hathor template override php layout

* Changes suggested by zero-24
* Add link to Joomla updater instructions for developer's benefit

* Revised version to reflect some of Brian's ideas

* Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

* Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

* Revised layout in traffic light colour sequence, using 'More Detail' and chevrons to indicate more detail available link and including introductory explanation

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update media/com_joomlaupdate/js/default.js

Co-authored-by: Quy <quy@fluxbb.org>

* Update media/com_joomlaupdate/js/default.js

Co-authored-by: Quy <quy@fluxbb.org>

* Language string tweak as per bayareajenn's suggestion

* First cut at flagging plugins that need attention

* Add language string, confirm dialog and messages

* should not have committed .gitignore

* phpcs correction

* Update administrator/components/com_joomlaupdate/models/default.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/components/com_joomlaupdate/models/default.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/components/com_joomlaupdate/views/default/tmpl/default_update.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/components/com_joomlaupdate/views/default/tmpl/default_update.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/components/com_joomlaupdate/views/default/tmpl/default_update.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* Update administrator/components/com_joomlaupdate/views/default/tmpl/default_update.php

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

* zero-24 suggestions

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/en-GB.com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Updates as suggested in Git Discussion to make potentially serious problem message clearer and to reduce prominence of php settings check adding in More Detail option

* Updates as suggested in Git Discussion

* Add problem plugin author and author URL information to the warning information
Fix for javascript error thrown when no no live update is available
Start update checker on DOMContentLoaded

* phpcs

* Fix for preupdatechecker not auto-starting automatically

* Change language string  COM_JOOMLAUPDATE_VIEW_DEFAULT_SHOW_MORE_COMPATIBILITY_INFORMATION to COM_JOOMLAUPDATE_VIEW_DEFAULT_EXTENSIONS_SHOW_MORE_COMPATIBILITY_INFORMATION so that it fits the relevant pattern of strings better
Change "Currently Compatible Version" and "Update Compatible Version" to refer to current and target Joomla versions (@particthistle)
Re-ordered columns -> "Extension Name, Extension Type, Installed Version, Joomla '3.x' Compatible Version, Joomla '4.x' Compatible Version" to make it easier to read and show progression of Joomla versions (@particthistle)
Hide plugin warning and table headings as soon as all the plugins have been tested if they are all 'safe' on the 'live update' tab (@zero-24)

* When there are no extensions installed or no non-Core Critical Plugins installed disable the wanrning messages and immediately exit the checking process.

Co-authored-by: Geraint Edwards <jevents@gwesystems.com>
Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: zero-24 <zero-24@users.noreply.github.com>
Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.0] fix webp support

#31048 added support for webp images however it was not complete. MediaHelper::isImage was not updated to support webp. As a result while webp could be added to media manager as an allowed image format everything worked except for the display of the thumbnail.

As this PR fixes that we might as well add webp to the default list of supported image formats.

To test

1. either do a fresh install with the prebuilt package OR apply this pr, run npm ci and go to the media manager options. delete all the values in the Legal Extensions (File Types), Legal Image Extensions (File Types) and Legal MIME Types and then save the options. This will reset the value to the new defaults.

2. Try to upload any webp image
3. Try to crop and resize the image
4. Try to rename the image
5. Get a shareable link and make sure it works

brianteeman committed:
[4.0] fix webp support

#31048 added support for webp images however it was not complete. MediaHelper::isImage was not updated to support webp. As a result while webp could be added to media manager as an allowed image format everything worked except for the display of the thumbnail.

As this PR fixes that we might as well add webp to the default list of supported image formats.

To test

1. either do a fresh install with the prebuilt package OR apply this pr, run npm ci and go to the media manager options. delete all the values in the Legal Extensions (File Types), Legal Image Extensions (File Types) and Legal MIME Types and then save the options. This will reset the value to the new defaults.

2. Try to upload any webp image
3. Try to crop and resize the image
4. Try to rename the image
5. Get a shareable link and make sure it works

sandramay0905 commented:
@brianteeman on a fresh install using launch.joomla.org i did a Upload & Update using a prebuilt package. 

Upload of webp-image works, not crop and resize as the changes are not saved. 

I didn't test further cause i didn't do the test instructions "do a fresh install with the prebuilt package OR apply this pr, run npm" and i guessed thats why the changes are not saved.

If my guess is wrong i can mark the test as unsuccessfully.

brianteeman commented:
> I didn't test further cause i didn't do the test instructions "do a fresh install with the prebuilt package OR apply this pr, run npm" and i guessed thats why the changes are not saved.

Without falling those test instructions it wont work

brianteeman committed:
[4.0] Cassiopeia unpublished badges

The markup for the Unpublished, Not Published Yet and Expired badges that is displayed in various views on the front end to administrators was incorrect.

### Before

### After

regularlabs committed:
[J4] btn-group-yesno: success color only applied to 1 value (#33091)

* [J4] btn-group-yesno: success color only applied to 1 value

When using the `btn-group-yesno` class on radio buttons in XML forms, the green `btn-outline-success` is applied to options with a value other than `0`.
This causes multiple buttons to be green.

This PR fixes that.

It also fixes the issue with the `disabled` attribute not being added to the options when using the `btn-group-yesno` class.

* Fixes class not being overridable

brianteeman committed:
Revert "[4.0] Cassiopeia unpublished badges"

This reverts commit 6d974127490183d2412db14acee1755cedd0b493.

brianteeman committed:
Revert "[4.0] Cassiopeia unpublished badges"

This reverts commit 6d974127490183d2412db14acee1755cedd0b493.

joomdonation committed:
[4.0] Remove unnecessary version compare (#33085)

* Remove unnecessary version compare

brianteeman committed:
[4.0] Default Menu item (#33124)

When setting the default menu item on and off you will see that the publish up and publish down fields are shown/hidden as appropriate. However because they are above the home button it creates a usability and an accessibility as you have already navigated passed those fields.

This PR simply changes the order.

Quy commented on administrator/components/com_media/config.xml:
```diff
@@ -9,7 +9,7 @@
 			description="COM_MEDIA_FIELD_LEGAL_EXTENSIONS_DESC"
 			size="50"
-			default="bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS"
+			default="bmp,csv,doc,gif,ico,jpg,jpeg,webp,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,WEBP,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS"
```
Values are listed in alpha order.

chmst committed:
Improve text for uploadinfo (#33116)

Commit:
Move markup to not display bottom border in sample data list (#33106)

brianteeman committed:
[4.0] Cassiopeia unpublished badges (#33102)

Commit:
[4] ERR_CONTENT_DECODING_FAILED help (#33097)

dgrammatiko committed:
[4.0] Fix non lazy loaded images (#33093)

* Fix non lazy loaded images

* Update media.php

infograf768 committed:
[4.0] Articles frontend associations badges changed to buttons and more (#32992)

sandewt committed:
[4.0] Improvement Vote Plugin (#31098)

CI error is unrelated

brianteeman committed:
Update config.xml

brianteeman committed:
Update config.xml

Commit:
Add period (#33132)

brianteeman committed:
[4.0] Sample Data div (#33134)

Not quite sure why it's there but the typography article is wrapped in a div that is not required

Quy commented on installation/sql/mysql/base.sql:
```diff
@@ -153,13 +153,13 @@ INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`,
 (0, 'com_languages', 'component', 'com_languages', '', 1, 1, 1, 1, 1, '', '{"administrator":"en-GB","site":"en-GB"}', ''),
 (0, 'com_login', 'component', 'com_login', '', 1, 1, 1, 1, 1, '', '', ''),
-(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg,png" ...
+(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,webp,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg ...
```
```suggestion
(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,txt,webp,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,WEBP,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg,png,webp","ignore_extensions":"","upload_mime":"image\\/jpeg,image\\/gif,image\\/png,image\\/bmp,image\\/webp,application\\/msword,application\\/excel,application\\/pdf,application\\/powerpoint,text\\/plain,application\\/x-zip","upload_mime_illegal":"text\\/html"}', ''),
```

Quy commented on installation/sql/postgresql/base.sql:
```diff
@@ -159,13 +159,13 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
 (0, 'com_languages', 'component', 'com_languages', '', 1, 1, 1, 1, 1, '', '{"administrator":"en-GB","site":"en-GB"}', '', 0, 0),
 (0, 'com_login', 'component', 'com_login', '', 1, 1, 1, 1, 1, '', '', '', 0, 0),
-(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg,png" ...
+(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,webp,odg,odp,ods,odt,pdf,png,ppt,txt,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg ...
```
```suggestion
(0, 'com_media', 'component', 'com_media', '', 1, 1, 0, 1, 1, '', '{"upload_extensions":"bmp,csv,doc,gif,ico,jpg,jpeg,odg,odp,ods,odt,pdf,png,ppt,txt,webp,xcf,xls,BMP,CSV,DOC,GIF,ICO,JPG,JPEG,ODG,ODP,ODS,ODT,PDF,PNG,PPT,TXT,WEBP,XCF,XLS","upload_maxsize":"10","file_path":"images","image_path":"images","restrict_uploads":"1","allowed_media_usergroup":"3","check_mime":"1","image_extensions":"bmp,gif,jpg,png,webp","ignore_extensions":"","upload_mime":"image\\/jpeg,image\\/gif,image\\/png,image\\/bmp,image\\/webp,application\\/msword,application\\/excel,application\\/pdf,application\\/powerpoint,text\\/plain,application\\/x-zip","upload_mime_illegal":"text\\/html"}', '', 0, 0),
```

Commit:
Standardize label (#33136)

rjharishabh committed:
[4.0] css changes in voting plugin (#32919)

* css changed in voting

* inlined

* width auto

* space removed

* dropdown added

* responsive

Commit:
s/form/from (#33147)

Commit:
Show cli command title before asking a question (#33146)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
[4] Change link to Joomla News (#33058)

brianteeman committed:
[4.0] TinyMCE editor.css (#33112)

Commit:
[4] show_total_votes in vote plugin (#33143)

Commit:
s/form/from (#33154)

Commit:
s/compatiblity/compatibility (#33157)

Commit:
s/hightlight/highlight (#33158)

Commit:
s/packgage/package (#33160)

Commit:
s/Listers/Listeners (#33161)

Commit:
s/docunment/document (#33162)

Commit:
s/calcuate/calculate (#33163)

Commit:
s/sipmle/simple (#33159)

brianteeman committed:
[4.0] tags menu items image alt [a11y] (#31359)

Commit:
return constant and not int (#33148)

Commit:
Move class constants from DateTime to DateTimeInterface (#33156)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
Hide Sample Data module if no Sample Data is available (#33168)

Commit:
Hide header links (#33140)

Commit:
unpublish stats plugin on selecting once or never (#33150)

Fedik committed:
[4.0] Joomla.request: callback on the request completed (#33099)

chmst committed:
[4.0] Blog Sampledata: Fix links, use  camelCase for variable catIds (#31711)

AndySDH committed:
[4.0] Updating all views to use H1 for Component Title when Menu Title is hidden (#32525)

* Update category_default.php

* Update list.php

* Remove unused page_subheading param, Remove unused page_subheading

* Update components/com_tags/src/View/Tag/HtmlView.php

* Update joomla.ini

* Update multilang.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Commit:
Fix markup (#33191)

wilsonge committed:
Merge commit '836f877' into 4.0-dev

wilsonge committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

brianteeman committed:
[4.0] Subforms/subfields (#32864)

Commit:
[4.0] Adding a manage button to the sample data module (#33169)

particthistle committed:
[4.0] Rename Web Services api-authentication plugins (See #31306) (#33174)

Commit:
[4] msg: You cannot add or remove authenticators on behalf of users (#33064)

Commit:
Add session handler to cli (#33145)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
[4] show result of checking for updates in CLI (#33153)

dgrammatiko committed:
[4.0] Decouple messages from core.js (#32747)

dgrammatiko committed:
[4.0] Remove the events wrapper from core.js (#32748)

dgrammatiko committed:
[4.0] highlight js (#33089)

dgrammatiko committed:
[4.0] Make  lighthouse happy (#32141)

I hope I got the comments and discussion right and problems are fixed so we are good to merge this as an improvement and example how someone can do it.

YatharthVyas committed:
[4.0] Fixes radio fields layout (#33176)

Kostelano committed:
Correct field layout in "User - Joomla API Token" (#33188)

wilsonge committed:
Merge commit 'de9e662' into 4.0-dev

Commit:
Correct docblock (#33184)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
Fix search tools alignment (#33204)

Hackwar committed:
Fixing component ID in testing sampledata (#33206)

richard67 committed:
[4.0] Adapt unit tests to PR 32718 having been merged up from 3.10 (#33207)

merged on review

particthistle committed:
[4.0] Improve success message for when you upload a file to a template (#33181)

rjharishabh committed:
click to grid-check (#33223)

YatharthVyas committed:
places delete button to the right of cancel (#33221)

rjharishabh committed:
[4.0] Change layout of filter search in articles (archive view) (#33219)

* changed div to span

* lines indented

brianteeman committed:
[4.0] Edit image modal toolbar (#33231)

Adds the correct class for the toolbar

### Before

### After

### To test
Select the Image XTD editor button from the editor
Click on an image
Click on the pencil icon to open the new view

rjharishabh committed:
[4.0] click to check the checkbox in com_finder&view=maps (#33228)

added behavior.multiselect

rjharishabh committed:
[4.0] click to check the checkbox in com_finder&view=index (#33227)

added behavior.multiselect

dgrammatiko committed:
[4.0] Xtd modals done right (#33216)

Pull Request for Issue #33210 .
- Modals should not break when debug language is on
- Modals should have unique ids (since there is an option for multiple instances of an editor)

Co-authored-by: Quy <quy@fluxbb.org>

rjharishabh committed:
[4.0] change layout of select in search (frontend) (#33220)

added class form-select, w-auto and mb-2

Commit:
New fatal error page (#30056)

joomdonation committed:
[4.0] Rename Fields - Subfields plugin to Fields - Subform plugin (#33200)

brianteeman committed:
[4.0] Admin modules bootstrap (#33045)

joomdonation committed:
[4.0] Mail Templates improvement (#33230)

YatharthVyas committed:
Rename location to file (#33248)

Commit:
CTRL + OPTION + A on MacOS. (#33254)

Commit:
[4] Hide empty stats when plugin disabled (#33261)

brianteeman committed:
[4.0] system css (#33258)

This PR removes the reference to background images that are not part of the release.

Code review should be fine

mrsgv committed:
[4.0]Changed the helper text in Redirect plugins Advanced settings (#32840)

* Changed redirect help message

* Removed the space.

dgrammatiko committed:
[4.0] J4 Modules should require an element field (#33182)

* J4 Modules require an element field
* Update ModuleAdapter.php

Commit:
[4] Remove use of deprecated triggerEvent call (#33166)

* Remove use of deprecated triggerEvent call

* Handle when all sample plugins are disabled

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
Add unit test for ModuleAdapter::getElement (#33260)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

chmst committed:
[4.0] Blog layout ux (#32943)

coolcat-creations committed:
Add Module Position above the logo (#31731)

Pradumn27 committed:
Fixes the Inconsistent placement of frontend action buttons (#32703)

joomdonation committed:
[4.0] Migrate data for media field in repeatable field (#33111)

brianteeman committed:
[4.0] webp support in templates (#33087)

brianteeman committed:
[4.0] XTD-Editor fields button disappears (#33259)

PhilETaylor committed:
[4] Introducing Joomla’s first blank state (#33264)

brianteeman committed:
[4.0] com_contact blank state (#33265)

brianteeman committed:
[4.0] com_newsfeeds blank state (#33268)

Commit:
[4.0] Move form text below field (#33275)

brianteeman committed:
[4.0] com_users notes blank state (#33273)

YatharthVyas committed:
[4.0] Rephrase SEND_FAILURE_BODY when registration fails (#33276)

brianteeman committed:
[4.0] com_banners blank state (#33269)

brianteeman committed:
[4.0] com_privacy requests and consents blank state (#33272)

brianteeman committed:
[4.0] com_messages blank state (#33274)

joomdonation committed:
[4.0] Subform Cleanup (#33262)

rjharishabh committed:
click to grid check (#33278)

PhilETaylor committed:
Use CMS library which is FTP Layer aware (#33283)

joomdonation committed:
[4.0] Subform UX (#33096)

PhilETaylor committed:
update JLIB_CLIENT_ERROR_FTP_BAD_RESPONSE (#33284)

PhilETaylor committed:
[4] Correct link to private messages from module (#33285)

richard67 committed:
Alpha-order use statements (#33297)

 code review

brianteeman committed:
[4.0] Edit profile icons (#33294)

Adds the missing icons to the save and cancel buttons on the front end edit profile page.

To test makes sure that the login module is set to display the profile link

Then on the front end login and seelct the profile and then the edit profile

Scroll down and you will see the two buttons without icons

Apply this pr and the icons are present

partial pr for #32250

brianteeman committed:
[4.0] Front end edit contact incorrect icon (#33295)

The incorrect (missing) icon was being used for the cancel button

To test make sure you have a contact form published and then when logged in as an admin click on the edit contact

Scroll down and you will see that the red cancel button has no icon

Apply this PR and the icon is present

brianteeman committed:
[4.0] Assign a color to Menu "Home" (#33298)

PR for #31980

There are different icons for the home and regular menu items but they are both grey in colour. This PR adds the same colour used for featured to the home icon so it stands out a little more.

YatharthVyas committed:
[4.0] com_postinstall: Display "Showing messages for" dropdown only if there's more than 1 options (#33300)

* check count before rendering dropdown

* fix conditional operator

* Remove redundant type cast

* Use CSS to hide form

Add `visually-hidden` class to the markup if options are not > 1

* Remove spac

Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Restyle the Post Installation No Messages page (Blank State Style) (#33287)

* Restyle the no messages page

* simplify the HTML

brianteeman committed:
[4.0] Sample Data menu permission (#33299)

The permission for the template settings menu item was set to public. This meant that as long as you could log in to the site in the front end then you would see the link to edit the template. However only a super user can do this so you got an ugly error page.

This PR correctly sets it to Super Users the same as with the Site settings menu itm

joomdonation committed:
Fix setError call (#33302)

Remove second unused param.

Commit:
[4.0] Moving reusable parts of blankstate layout to a JLayout (#33288)

Pull Request for Issue #33279 .

- Moved the repeating parts of the layout to a JLayout which can be used also by 3rd parties.
- Added permissions checks for the "Add xxx" button. The permissions checks are kept as similar as reasonable to the one for the toolbar button
- Corrected the link of the "Add xxx" button to point to the respective controller task instead of directly going to the view.
- Adjusted the language strings for the "_CONTENT" so they only differate in the prefix. This could also be changed to "_ITEMS" but I think basically everything is "content" of some sort as well, not only com_content items. In the end, the name of the string doesn't matter anyway much.
- Move the "Learn More" string to the joomla.ini file, because this one always is the same.

Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Quy <quy@fluxbb.org>

drmenzelit committed:
Remove padding between module positions (#33310)

bembelimen committed:
Fix dropdown hide when workflow is active (#33313)

bembelimen committed:
Fix wrong events (#33316)

brianteeman committed:
[4.0] s/an/a (#33318)

Corrects an english language typo

brianteeman committed:
[4.0] s/an/and (#33320)

Typo in comment

joomdonation committed:
Set only_use_in_subform = 1 for subfields migrated from repeatable fields (#33301)

PhilETaylor committed:
[4] Allow in-memory autoload_psr4.php (#33263)

PhilETaylor committed:
[4] Remove release news from post installation messages screen (#33324)

* Remove release news from post installation messages screen

* Update administrator/components/com_postinstall/tmpl/messages/default.php

Co-authored-by: Quy <quy@fluxbb.org>

* remove COM_POSTINSTALL_LBL_RELEASENEWS

* remove deprecated method call and revert removal of FActory

Co-authored-by: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] package typo (#33329)

Fix copy paste error on the package details

PhilETaylor committed:
[4] Remove deprecated function calls and use short array syntax (#33322)

brianteeman committed:
[4.0] Toolbar Close/Cancel (#33317)

* [4.0] Toolbar Close/Cancel

The cancel button should have the text
Cancel when its a new item
Close when its an existing item

With this PR I have fixed the few instances that incorrectly did not follow this pattern which is used in all the other 33 views.

Summary of changes
Create a New Article - Cancel
Edit an Article - Close
Create a Workflow - Cancel
Edit a Workflow - Close
Create a Workflow Stage - Cancel
Edit a Workflow Stage - Close
Create a Workflow Transition - Cancel
Edit a Workflow Transition - Close
Edit an image - Close
Create a private message - Close
Read a private message - Close

* CS

* space

* article

* space

PhilETaylor committed:
Add icons to blank state (#33333)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

Commit:
Don't show the create button if in a modal (#33331)

rjharishabh committed:
Add path (#33330)

Commit:
Add png to default values (#33345)

PhilETaylor committed:
cs /plugins (#33354)

Commit:
Style access checkbox. (#33366)

Commit:
Remove end margin (#33349)

rjharishabh committed:
[4.0] Add bottom-margin in vote (#33340)

YatharthVyas committed:
[4.0] Postinstall reset hide fix (#33338)

* Fix Hide All Messages

The $eid was always assigned 212 (ID of Joomla CMS) by ` $model->getState('eid', $model->getJoomlaFilesExtensionId())`
Now, this value is dynamically assigned based on the value of `#adminForm`'s eid input field (extension options dropdown field)

* fix reset messages

$eid was always assigned the value of 212 (ID of Joomla CMS) by `getJoomlaFilesExtensionId())` Now it dynamically takes the id from the URL parameters.
Also, removed line 39 as the PHPCS suggested I do so.

* update get to getInt

PhilETaylor committed:
[4] Rename recent BlankState feature to EmptyState (#33334)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>
Co-authored-by: Quy <quy@fluxbb.org>

joomdonation committed:
[4.0] Field contexts filter improvement (#33387)

* Field contexts filter improvement

* CS

PhilETaylor committed:
s/editor/textarea (#33395)

brianteeman committed:
[4.0] pre-update check (#33393)

when moving from bs2 to bs5 the label-* classes were changed to alert-* classes

however there is no such class as alert-important - it should be alert-danger

YatharthVyas committed:
encloses account details and 2fa form in fieldset (#33389)

brianteeman committed:
[4.0] Sample Data focus ring (#33407)

Adds some margin so that the entire focus ring is visible. (Usually seen when you navigate using the keyboard)

### Before

### After

PhilETaylor committed:
Allow newsfeed creation if no categories (#33405)

drmenzelit committed:
[4.0] Set color for text and links in position topbar (#33394)

* Set color for text and links in position topbar

* Add empty line

Kostelano committed:
[4] [mod_tags_similar] Transferring the class from DIV to UL (#33385)

* Removed unused constants

* Update com_workflow.ini

* Transferring the class from DIV to UL

brianteeman committed:
[4.0] Empty Trash Button (#33402)

When you have filtered your items by the trashed status in most components there is an empty trash button on the toolbar. However for contacts, newsfeeds, redirects and tags the button is in the actions dropdown.

For a consistent UI they should all be the same.

To test apply the pr
filter a component by trashed and observe the button

PhilETaylor committed:
[4] Generic Blank State for Categories of Extensions & Tags (#33286)

* Blank State for Categories of Extensions

Allow more generic use of filering by where

Further refine use of filtering

Add Tags Blank State

Refine the lookup with a non-generic filter

dont use input use filter

update copyright date

Co-authored-by: Brian Teeman <brian@teeman.net>

update copyright date

Co-authored-by: Brian Teeman <brian@teeman.net>

refactor due to upstream changes

Refactor due to upstream change

convert array to short syntax

correct url

remove unused lang

Co-authored-by: Quy <quy@fluxbb.org>

Correct help url

add really unhelpful documentation link

reindex array

remove indent

Special handling for the title as com_categories is a service component

Update administrator/components/com_categories/tmpl/categories/blankstate.php

add correct icon

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

add link for learn more button

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

revert

Update administrator/components/com_tags/tmpl/tags/blankstate.php

Co-authored-by: Quy <quy@fluxbb.org>

Align =>

empty state rename from blank slate

* Update administrator/components/com_categories/src/Model/CategoriesModel.php

* Update administrator/components/com_tags/src/Model/TagsModel.php

* remove deprecated calls and ensure only new button shown on empty state

* Remove spaces

* Prepared statement

* make private

Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Blank State for com_cache (#33325)

* add com_cache blank slate

* Dont display toolbar if no cache items

* Add blank state language strings

* namespace \count function calls

* /s/you/if

* Update administrator/components/com_cache/tmpl/cache/blankstate.php

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/components/com_cache/tmpl/cache/blankstate.php

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update administrator/language/en-GB/com_cache.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

* Add icon

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

* a-z lang strings

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

* s/blankslate/emptystate

* Short array syntax

* Update administrator/language/en-GB/com_cache.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Empty State for Banner clients and tracks (#33381)

* Empty State for Banner clients and tracks

* Remove formURL

Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Global Check-in Empty State (#33383)

* Global Check-in Empty State

* remove language string not used

* tabs

* change text

* cs

* remove formURL

Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Empty state for com_redirect (#33384)

* Empty state for com_redirect

* merge from other pr

* change icon

* cs

* alpha

* Update administrator/language/en-GB/com_redirect.ini

Co-authored-by: Quy <quy@fluxbb.org>

* Update administrator/components/com_redirect/tmpl/links/emptystate.php

Co-authored-by: Brian Teeman <brian@teeman.net>

Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: Brian Teeman <brian@teeman.net>

PhilETaylor committed:
com_installer warnings empty state (#33411)

PhilETaylor committed:
com_csp empty state (#33416)

PhilETaylor committed:
[4] Remove disabled and useless buttons on empty state leaving just New (#33380)

* Remove disabled and useless buttons on empty state leaving just New

* Hide empty trash if last status filtered was trashed

* Update administrator/components/com_tags/src/View/Tags/HtmlView.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.0] TinyMCE plugin layout (#33177)

* [4.0] (WIP) TinyMCE plugin layout

There have been multiple issues raised about the UI of the tinymce plugin and the lack of clarity that the options are for different sets. It's also a very regular topic on the forum whereby users believe they have configured the plugin and it doesn't work. When the problem is that they are simply in the wrong set.

This is a work in progress attempt to improve this.

The basic concept of this change it to move the "All available menus and buttons." inside each set so that the tabs are at the top.

The other main change (which is not done yet) is to move the "Assign this Set to" field and label to the very first field in the tab.

Hopefully this will make it much much more obvious that you are completing a set of options for a usergroup and not for the entire plugin.

(If anyone has a suggestion how to display that field at the very top it would be appreciated)

* access field

* draft

* date

brianteeman committed:
[4.0] Media Manager - Can't preview entire image (#33323)

* [4.0] Media Manager - Can't preview entire image

Pull request for #29149

This requires npm ci as it's a css change

### Steps to reproduce the issue
- Download https://unsplash.com/photos/xnEZYWmfPwU/download?force=true (original size)
- Upload to the Media Manage or place in your images directory
- Double click the image to preview it

### Before
Half the image is cropped and you cannot scroll down to view the other half

### After
The entire image is visable

* tabs

* lint

* portrait

brianteeman committed:
[4.0] regenerate helptoc (#33415)

Runs the buildTOC.php script. There is at least one new entry when you go to the help page Action Logs

PhilETaylor committed:
[4] Fix trash menu items displaying logic (#33406)

Fix trashing items ACL checks

PhilETaylor committed:
[4] Allow banner creation if there are not categories (#33399)

* add ACL check

* allow adding banners from empty state if no categories

Kostelano committed:
[4] [Workflow] Removed unused constants (#33377)

PhilETaylor committed:
Only show versions if the article has been saved at least once (#33412)

PhilETaylor committed:
[4] When item doesn't exist, show specific message (#33414)

brianteeman committed:
[4.0] icon for non featured (#33417)

richard67 committed:
[4.0] Add update SQL script for admin modules parameters changed by PR #33045 (#33375)

rjharishabh committed:
[4.0] click to check the checkbox in com_cache (#33413)

* class-attribute two to one

* add previous label

brianteeman committed:
[4.0] behaviour plugin (#33420)

change the description of the behaviour plugins to remove the word "content" to avoid confusion with the word "articles"

brianteeman committed:
[4.0] user groups - col headers (#33423)

* [4.0] user groups - col headers

There really is no need to hide the column header text here - there is more than enough space in the view

* remove duplicate title

brianteeman committed:
[4.0] Debug Group scope (#33425)

The attribute scope is essential to tell users of assistive technology if the header applies to the column or the row. However that means it can only be applied to a table header (th) and not to a table cell (td).

There is a small cosmetic change as a result in that the contents of the cell are now visuall displayed in bold.

To test go to users-> groups and click on the permissions icon.

All the content in the first column should be in a th with a scope of row and visually displayed as bold

sandewt committed:
Move the dif tag (#33426)

Move the div tag

brianteeman committed:
[4.0] Modules modal scope (#33427)

The attribute scope is essential to tell users of assistive technology if the header applies to the column or the row. However that means it can only be applied to a table header (th) and not to a table cell (td).

There is a small cosmetic change as a result in that the contents of the cell are now visually displayed in bold.

Easiest way to test is to try and insert a module into an article. The module modal will be displayed

All the content in the first column should be in a th with a scope of row and visually displayed as bold

Commit:
Rename alert-default to alert-secondary (#33429)

There is no "alert-default" in BS5. Rename it to its equivalency.

infograf768 committed:
[4.0] Language switcher: improving display of dropdown with flags (#33082)


Co-authored-by: Brian Teeman <brian@teeman.net>

rjharishabh committed:
remove attributes (#33469)

brianteeman committed:
[4.0] privacy request scope (#33461)

The attribute scope is essential to tell users of assistive technology if the header applies to the column or the row. However that means it can only be applied to a table header (th) and not to a table cell (td).

There is a small cosmetic change as a result in that the contents of the cell are now visually displayed in bold.

To test go to users-> privacy requests and create a new request and then save

All the content in the Email column should be in a th with a scope of row and visually displayed as bold

Kostelano committed:
Removed <hr> tag in installation plugins (#33462)

Commit:
Hide columns in mobile view (#33458)

This is the same behavior as Menus > Manage to hide columns in mobile viewport.

Fedik committed:
Fix tinymce the options form (#33464)

rjharishabh committed:
[4.0] Separable buttons on Joomla Update Page (#33446)

* remove btn-group

* remove blank lines

* indent

YatharthVyas committed:
[4.0] com_templates: Validate input for copy template form (#33444)

* add required attribute to input

* add invalid template name in language

* display error message

* revert template error message

* use existing language constant for error message

* Update administrator/components/com_templates/src/Controller/TemplateController.php

Co-authored-by: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
[4] Add intl extension check to system info, and while there clean up deprecations/types/[] (#33457)

* add check for intl and moderise code and remove deprecated calls

* php modernisation

* HArd code the names of PHP Extensions to avoid them being translated

rjharishabh committed:
add class options-form (#33441)

YatharthVyas committed:
[4.0] Private Messages: Add legend in Configuration Form (#33442)

* add fieldset and legend in com_messages

* remove unused language constant

Ran a global search using IDE and couldn't find this being used anywhere in the entire repository

* add new constant for configuration form

* update legend text

* arrange imports in lexicographic order

* rename constant

* rename constant in legend text

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

PhilETaylor committed:
[4] Cleanup output html of email field (#33475)

Cleanup output of email field

brianteeman committed:
[4.0] toggle editor icon (#33470)

Pull Request for Issue #33465
Change the icon when the toggle editor button is toggled

wilsonge committed:
Merge commit '60260bb' into 4.0-dev

wilsonge committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev

richard67 committed:
Fix merge conflict remainder from upmerge (#33478)

richard67 committed:
PHPCS fixes after upmerge (#33479)

wilsonge committed:
JS CS

brianteeman committed:
caps

brianteeman committed:
caps

brianteeman committed:
Merge remote-tracking branch 'origin/webpmedia' into webpmedia

brianteeman committed:
sort

brianteeman commented:
closed - will make new pr
